actors/builtin/multisig/multisig_test.go

```diff
@@ -637,6 +637,163 @@ func TestApprove(t *testing.T) {
+
+		// anne proposes a transaction
+		rt.SetCaller(anne, builtin.AccountActorCodeID)
+		rt.ExpectValidateCallerType(builtin.AccountActorCodeID, builtin.MultisigActorCodeID)
+		actor.proposeOK(rt, chuck, sendValue, fakeMethod, fakeParams, nil)
+		rt.Verify()
+
+		// bob approves the transaction (number of approvals is now two but threshold is three)
+		rt.SetCaller(bob, builtin.AccountActorCodeID)
+		rt.ExpectValidateCallerType(builtin.AccountActorCodeID, builtin.MultisigActorCodeID)
+		proposalHashData := makeProposalHash(t, &multisig.Transaction{
+			To:       chuck,
+			Value:    sendValue,
+			Method:   fakeMethod,
+			Params:   fakeParams,
+			Approved: []addr.Address{anne},
+		})
+
+		actor.approveOK(rt, txnID, proposalHashData, nil)
+		rt.Verify()
+
+		// reduce the threshold so the transaction is already approved
+		rt.SetCaller(receiver, builtin.MultisigActorCodeID)
+		rt.ExpectValidateCallerAddr(receiver)
+		actor.changeNumApprovalsThreshold(rt, newThreshold)
+		rt.Verify()
+
+		// even if bob calls for an approval again(duplicate approval), transaction is executed because the threshold has been met.
+		rt.ExpectSend(chuck, fakeMethod, fakeParams, sendValue, nil, 0)
+		rt.SetBalance(sendValue)
+		rt.SetCaller(bob, builtin.AccountActorCodeID)
+		rt.ExpectValidateCallerType(builtin.AccountActorCodeID, builtin.MultisigActorCodeID)
+		actor.approveOK(rt, txnID, proposalHashData, nil)
+		rt.Verify()
+
+		// Transaction should be removed from actor state after send
+		actor.assertTransactions(rt)
+	})
+
+	t.Run("approve transaction if number of approvers has already crossed threshold", func(t *testing.T) {
+		rt := builder.Build(t)
+		const newThreshold = 1
+		signers := []addr.Address{anne, bob}
+		actor.constructAndVerify(rt, numApprovals, noUnlockDuration, signers...)
+
+		// anne proposes a transaction
+		rt.SetCaller(anne, builtin.AccountActorCodeID)
+		rt.ExpectValidateCallerType(builtin.AccountActorCodeID, builtin.MultisigActorCodeID)
+		actor.proposeOK(rt, chuck, sendValue, fakeMethod, fakeParams, nil)
+		rt.Verify()
+
+		// reduce the threshold so the transaction is already approved
+		rt.SetCaller(receiver, builtin.MultisigActorCodeID)
+		rt.ExpectValidateCallerAddr(receiver)
+		actor.changeNumApprovalsThreshold(rt, newThreshold)
+		rt.Verify()
+
+		// bob attempts to approve the transaction but it gets approved without
+		// processing his approval as it the threshold has been met.
+		rt.ExpectSend(chuck, fakeMethod, fakeParams, sendValue, nil, 0)
+		rt.SetBalance(sendValue)
+		rt.SetCaller(bob, builtin.AccountActorCodeID)
+		rt.ExpectValidateCallerType(builtin.AccountActorCodeID, builtin.MultisigActorCodeID)
+		proposalHashData := makeProposalHash(t, &multisig.Transaction{
+			To:       chuck,
+			Value:    sendValue,
+			Method:   fakeMethod,
+			Params:   fakeParams,
+			Approved: []addr.Address{anne},
+		})
+
+		actor.approveOK(rt, txnID, proposalHashData, nil)
+		rt.Verify()
```

Comment:
Nit: this `Verify` should really appear immediately after the `rt.Call` inside `approve`. In a follow-up PR, could you please make sure that every `Call` is immediately verified, and then remove unnecessary verifies from the test body code.

Reply:
Will do.